src/Weaviate.Client.Tests/Integration/TestProperties.cs

```diff
@@ -6,6 +6,224 @@ namespace Weaviate.Client.Tests.Integration;
+            new { testIntArray = new[] { 1, 2, 3 } },
+            "testIntArray",
+            new[] { 1L, 2L, 3L },
+        };
+        yield return new object[]
+        {
+            new[] { Property.BoolArray("testBoolArray") },
+            new { testBoolArray = new[] { true, false } },
+            "testBoolArray",
+            new[] { true, false },
+        };
+        yield return new object[]
+        {
+            new[] { Property.NumberArray("testNumberArray") },
+            new { testNumberArray = new[] { 1.1, 2.2, 3.3 } },
+            "testNumberArray",
+            new[] { 1.1, 2.2, 3.3 },
+        };
+        yield return new object[]
+        {
+            new[] { Property.NumberArray("testNumberArrayFloat") },
+            new { testNumberArrayFloat = new[] { 1.1f, 2.2f, 3.3f } },
+            "testNumberArrayFloat",
+            new[] { 1.1, 2.2, 3.3 },
+        };
+        yield return new object[]
+        {
+            new[] { Property.DateArray("testDateArray") },
+            new
+            {
+                testDateArray = new[] { new DateTime(2025, 10, 29), new DateTime(2025, 10, 30) },
+            },
+            "testDateArray",
+            new[] { new DateTime(2025, 10, 29), new DateTime(2025, 10, 30) },
+        };
+        yield return new object[]
+        {
+            new[] { Property.UuidArray("testUuidArray") },
+            new
+            {
+                testUuidArray = new[]
+                {
+                    Guid.Parse("12345678-1234-5678-1234-567812345678"),
+                    Guid.Parse("87654321-4321-8765-4321-876543218765"),
+                },
+            },
+            "testUuidArray",
+            new[]
+            {
+                Guid.Parse("12345678-1234-5678-1234-567812345678"),
+                Guid.Parse("87654321-4321-8765-4321-876543218765"),
+            },
+        };
+        // Anonymous types don't work well for the assertions below. Commenting them out for now.
+        // yield return new object[]
+        // {
+        //     new[]
+        //     {
+        //         Property.Object(
+        //             "testObject",
+        //             subProperties: new[] { Property.Text("testText"), Property.Int("testInt") }
+        //         ),
+        //     },
+        //     new { testObject = new { testText = "nested text", testInt = 99 } },
+        //     "testObject",
+        //     new { testText = "nested text", testInt = 99 },
+        // };
+        // Add more property types as needed
+    }
+
+    [Theory]
+    [MemberData(nameof(PropertyTestData))]
+    public async Task Property_OfAnyType_Should_SaveAndRetrieve(
+        Property[] props,
+        object obj,
+        string propertyName,
+        object expected
+    )
+    {
+        var c = await CollectionFactory(
+            description: $"Testing property {propertyName}",
+            properties: props
+        );
+
+        var id = await c.Data.Insert(obj);
```

Comment:
could you also test batch inserts so we're sure GRPC works?

Reply:
You can also just add a `var id2 = await c.Data.InsertMany([obj]);` and then add

```
for in in id, id2:
  ..current validation...
```

(sorry for the python/c# mix 😅 )